pkg/tools/etcd_tools.go

```diff
@@ -359,10 +409,11 @@ func (h *EtcdHelper) AtomicUpdate(key string, ptrToType runtime.Object, ignoreNo
 			return nil
 		}
 
-		_, err = h.Client.CompareAndSwap(key, string(data), 0, origBody, index)
+		response, err := h.Client.CompareAndSwap(key, string(data), 0, origBody, index)
```

Comment:
Since generic/etcd is essentially a rest store, I'd say the rest semantics are not satisfied for blind put, but all other storage level things are resolved.